pkgs/leak_tracker_flutter_testing/lib/src/leak_tracking_for_tests.dart

```diff
@@ -0,0 +1,183 @@
+
+  /// Creates copy of current settings to debug notGCed leaks.
+  static LeakTrackingForTestsSettings withCreationStackTrace() {
+    return settings.copyWith(
+      leakDiagnosticConfig:
+          const LeakDiagnosticConfig(collectStackTraceOnStart: true),
+    );
+  }
+
+  /// Creates copy of current settings to debug notDisposed leaks.
+  static LeakTrackingForTestsSettings withDisposalStackTrace() {
+    return settings.copyWith(
+      leakDiagnosticConfig:
+          const LeakDiagnosticConfig(collectStackTraceOnDisposal: true),
+    );
+  }
+
+  /// Creates copy of current settings, that collects retaining path for not GCed objects.
+  static LeakTrackingForTestsSettings withRetainingPath() {
+    return settings.copyWith(
+      leakDiagnosticConfig:
+          const LeakDiagnosticConfig(collectRetainingPathForNotGCed: true),
+    );
+  }
+
+  /// Adds certain classes and leak types to skip lists in [settings].
+  static void skip({
+    Map<String, int?> notGCed = const {},
+    bool allNotGCed = false,
+    Map<String, int?> notDisposed = const {},
+    bool allNotDisposed = false,
+    List<String> classes = const [],
+  }) {
+    settings = withSkipped(
+      notDisposed: notDisposed,
+      notGCed: notGCed,
+      allNotDisposed: allNotDisposed,
+      allNotGCed: allNotGCed,
+      classes: classes,
+    );
+  }
+
+  /// Returns copy of [settings] with extended skip lists.
+  ///
+  /// In the result the skip limit for a class is maximum of two original skip limits.
+  /// Items in [classes] will be added to all skip lists.
+  static LeakTrackingForTestsSettings withSkipped({
+    Map<String, int?> notGCed = const {},
+    bool allNotGCed = false,
+    Map<String, int?> notDisposed = const {},
+    bool allNotDisposed = false,
+    List<String> classes = const [],
+  }) {
+    Map<String, int?> addClassesToMap(
+      Map<String, int?> map,
+      List<String> classes,
+    ) {
+      return {...map}..addEntries(classes.map((c) => MapEntry(c, null)));
+    }
+
+    return settings.copyWith(
+      leakSkipLists: LeakSkipLists(
+        notGCed: settings.leakSkipLists.notGCed.merge(
+          LeakSkipList(
+            byClass: addClassesToMap(notGCed, classes),
+            skipAll: allNotGCed,
+          ),
+        ),
+        notDisposed: settings.leakSkipLists.notGCed.merge(
+          LeakSkipList(
+            byClass: addClassesToMap(notDisposed, classes),
+            skipAll: allNotDisposed,
+          ),
+        ),
+      ),
+    );
+  }
+
+  /// Returns copy of [settings] with reduced skip lists.
+  ///
+  /// Items in [classes] will be removed from all skip lists.
+  static LeakTrackingForTestsSettings withTracked({
+    List<String> notGCed = const [],
+    List<String> notDisposed = const [],
+    List<String> classes = const [],
+  }) {
+    final result = settings.copyWith(
+      leakSkipLists: LeakSkipLists(
+        notGCed: settings.leakSkipLists.notGCed.track(notGCed).track(classes),
```

Comment:
how about `.track([...notGCed, ...classes])` so we remove a call to `track`? same below.